### ISSUE TITLE
Azure Static Web App build

### ISSUE THREAD

**Issue by kendrickcc** [victoriadrake/hugo-theme-introduction#334]
I really like this theme and was devastated when I could not deploy to Azure using GitHub Actions.

Per the README, a requirement is Hugo extended with npm postcss installed. Default Azure Static Web App does not install. If following the standard process to deploy.

Default workflow: 
```
jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v3
        with:
          submodules: true
      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_MANGO_SKY_0950B4610 }}
          repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
          action: "upload"
          ###### Repository/Build Configurations - These values can be configured to match your app requirements. ######
          # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
          app_location: "/" # App source code path
          api_location: "" # Api source code path - optional
          output_location: "public" # Built app content directory - optional
          app_build_command: ""
          ###### End of Repository/Build Configurations ######
```
Standard message is: 
```
Error: Error building site: POSTCSS: failed to transform "css/main.css" (text/css). Check your PostCSS installation; install with "npm install postcss-cli"
```
Solution: 
Modify workflow to add actions to extend Hugo and install required npm modules. I went back and forth with node.js and running Hugo. In the end, running Hugo --minify was needed for the site to rebuild. Don't understand why, but when I don't use those lines, I'm back to the failed postcss error.
```
jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v3
        with:
          submodules: true

      - name: Setup Hugo
        uses: peaceiris/actions-hugo@v2.5.0
        with:
          hugo-version: 'latest'
          extended: true

      - name: Setup Node.js environment
        uses: actions/setup-node@v3

      - name: Install npm pre-reqs
        run: |
          npm install -g postcss postcss-cli autoprefixer
          npm fund

      - name: Run Hugo
        run: hugo --minify

      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_MANGO_SKY_0950B4610 }}
          repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
          action: "upload"
          ###### Repository/Build Configurations - These values can be configured to match your app requirements. ######
          # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
          app_location: "/" # App source code path
          api_location: "" # Api source code path - optional
          output_location: "public" # Built app content directory - optional
          app_build_command: ""
          ###### End of Repository/Build Configurations ######
```

Hope this helps anyone trying to use this theme in Azure using Static Web App.

I did try this initially in Netlify and got the same error. I'm assuming the same fix would work there if using GitHub actions to deploy.

**Comment by jenkuntz** [comment]
Hi @kendrickcc ... I'm running into the same issue but not solving even with following what you posted above. Did you need to add scripts to your package.json file as well to make this work?

**Comment by kendrickcc** [comment]
I don't recall editing the package.json, and I'm not able to reproduce it successfully. I will keep trying, but it appears I'm broken now as well.

**Comment by jenkuntz** [comment]
Sorry for the belated response, I should have posted this sooner. I did figure it out. This might not be the most technically correct explanation, but how I understand it is the Github Action Azure Static Web Apps creates does not support installing Hugo Extended. So, in order to work around it, I ended up altering the Build and Deploy section as the screenshot shows:
- change app_location to /public
- comment out the output_location
- add skip_app_build: true to Azure does not try to build the site again. 

Essentially GitHub Actions is building the site already, so I don't need Azure to do it again, when I can just push the published files to Azure instead. 

Hopefully this helps someone... !

![Hugo - github workflow edits](https://user-images.githubusercontent.com/44136928/226118290-0a53a89d-975f-4809-9ddc-adf875b70c5d.png)
